reflection.md

```diff
@@ -2,17 +2,71 @@
 
 ## Which operators were easiest to implement and why?
 
+The `||` and `~` operators were easiest to implement (after figuring out the
+implicit conversion code, see below) since they closely resembled the sample we
+did in class with `add`/`+`. Their implementation was also a simple one-to-one call
+on the predefined case classes which contributed to their level of ease.
+
 ## Which operators were most difficult to implement and why?
 
+Although not directly an operator, the implicit conversion code (`string2regularExpression`
+and `char2literal`) was difficult to implement because it required
+putting the code in the appropriate *companion* class. At first, I had put
+`char2literal` in a `Literal` companion object, but it required me to add type
+annotations in `Program.scala` which seemed unnatural. After experimenting with
+moving the code up in the object inheritance hierchy, this problem was resolved.
+
+`string2regularExpression` came naturally after but was tricky to decide on the
+basecase for the fold: should it be `EMPTY` or `EPSILON`? After creating a buggy
+implementation with `EMPTY`, the distinction between `EMPTY` being a pattern
+and `EPSILON` being a character came back from 81! Debugging this was a bit tricky
+since it compiled but failed tests in unexpected places.
+
+`<*>` and `<+>` being postfix operators were difficult to figure out the syntax
+for. I had to recall that they (1) do not take an argument and (2) have `this`
+available in their implementation scope since they are defined on the `RegularExpression`
+class itself.
+
+The `{n}` (repetition operator) implementation came naturally, but figuring out
```

Comment:
Yes, it definitely took me a while to find out that there was a built-in 'apply' method.